packages/netlify-cms-core/src/formats/frontmatter.js

```diff
@@ -72,7 +72,7 @@ class FrontmatterFormatter {
     const result = matter(content, { engines: parsers, ...format });
     return {
       ...result.data,
-      body: result.content,
+      ...(result.content.trim() && { body: result.content }),
```

Comment:
@barthc do you mind sharing the reason this is required?

Reply:
It's the same line break issue at the end of the file. So when the file is parsed, it includes a `body` key with line break as value, this would occur for config fields without the top-level markdown body widget that resolves to the body of the markdown.

Reply:
Thanks I think I get it now.
We pass in an empty body here:
https://github.com/netlify/netlify-cms/blob/3e873f3e0277bca8f5dee619e931a66d47779c3d/packages/netlify-cms-core/src/formats/frontmatter.js#L80

when it doesn't exist, so we have to remove that property on parse.